simple_mediation.rmd

```diff
@@ -0,0 +1,123 @@
+```{r, include=FALSE}
+options(tinytex.verbose = TRUE)
+```
+
+```{r}
+#RQ: Does domain-specifical movement behaviour profile membership mediate the relationship between socioeconomic position and socio-emotional outcomes in children?
+##Simple mediation for males
+```
+
+```{r include=FALSE}
+library(medflex)
+library(dplyr)
+library(haven)
+library(rmarkdown)
+```
+
+```{r}
+# Load male dataset from Study 2, LSAC wave 6 data, and LSAC wave 8 data
+load(file = "Z:/LSAC dataset/Study_2/Study_2/df_males_domsp_weekday.RData")
+lsac_wave6 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb10.sas7bdat")
+lsac_wave8 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb14.sas7bdat")
+```
+
+```{r}
+# Select relevant columns from male dataset (hicid, profile membership, age, remoteness, Indigenous status, SEP)
+relevant_male_data <- df_male_domsp_weekday %>%
+  select(hicid, model3_trajectory_assignments, Age, Remoteness, Indigenous, SEP)
+
+# Select relevant columns from LSAC wave 6 (SEIFA varialbes), parents living at home, parent mental health, parent ethnicity)
+wave_6_relevant_data <- lsac_wave6 %>%
+  select(hicid, fcnfsad2d, fcnfsda2d, fcnfser2d, fcnfseo2d, fmoth, ffath, fsc13a1i, zf09fm, zf09ff) 
+
+# Update column names to make sense
+colnames(wave_6_relevant_data)[2:10] <- c("adv", "dis", "resources", "education_occupation", "mother_home", "father_home", "mental_health", "mother_race", "father_race")
+
+# Select relevant columns from LSAC Wave 8 (SDQ results)
+wave_8_relevant_data <- lsac_wave8 %>%
+  select(hicid, hapsoc:hasdqtb) 
+
+# Update column names to make sense
+colnames(wave_8_relevant_data)[2:7] <- c("prosocial", "hyperactivity", "emotional", "peer", "conduct", "total")
+
+# Combine columns to make new dataset
+domsp_wave6 <- inner_join(relevant_male_data, wave_6_relevant_data, by = "hicid")
+df_males <- inner_join(domsp_wave6, wave_8_relevant_data, by = "hicid")
+``` 
+
+```{r}
+#Take out participants with missing covariate data 
+colSums(df_males < -8)
+# 2 mental health, 9 mother race, 125 father race
+df_males <- df_males[!(df_males$mental_health == -9), ]
+df_males <- df_males[!(df_males$mother_race == -9), ]
+df_males <- df_males[!(df_males$father_race == -9), ]
+```
+
+```{r}
+# Check dataframe for missing data reported as N/A
+colSums(is.na(df_males))
+# 11 participants are missing SDQ data - this accounts for <1% of data and this information is essential for data anlaysis - therefore these participants will be removed from the dataset
+df_males_na_removed <- na.omit(df_males)
+```
+
+```{r}
+#Reduce mother and father race to 1 digit and convert to factor 
+df_males_na_removed$mother_race <- substr(df_males_na_removed$mother_race, 1, 1) %>% 
+  as.factor()
+df_males_na_removed$father_race <- substr(df_males_na_removed$father_race, 1, 1) %>% 
```

Comment:
A quick look at the tabs for this shows that some of these groups are still really tiny. E.g., group 9 has only 3 participants. I would have a look at if you can 1) collapse some of these (e.g., is the distinction between different areas of Asia important?) and 2) aggregate the smaller ones into an 'other' category. Depending on the reason you've included this, it might even be better just to go with born in aus vs not?